webrender/src/tiling.rs

```diff
@@ -415,17 +415,36 @@ fn add_to_batch(
         PrimitiveKind::Border => {
             let border_cpu =
                 &ctx.prim_store.cpu_borders[prim_metadata.cpu_prim_index.0];
-            // TODO(gw): Select correct blend mode for edges and corners!!
             let corner_kind = BatchKind::Transformable(
                 transform_kind,
                 TransformBatchKind::BorderCorner,
             );
-            let corner_key = BatchKey::new(corner_kind, blend_mode, no_textures);
+            // Selecting alpha blend for all corners is conservative,
+            // but no worse than what we previously had.
```

Comment:
s/alpha/premultiplied alpha/
also, why is this no worse?

Reply:
Before this patch every border was considered transparent, and so would always end up in the alpha pass for all segments.